Cubical/Categories/Instances/ZFunctors.agda

```diff
@@ -0,0 +1,412 @@
+    path = [ 1 , replicateFinVec 1 1r ]
+         ≡[ i ]⟨ [ 1 , replicateFinVec 1 (φ .snd .pres1 (~ i)) ] ⟩
+           [ 1 , replicateFinVec 1 (φ .fst 1r) ]
+         ≡[ i ]⟨ [ 1 , (++FinRid {n = 1} (replicateFinVec 1 (φ .fst 1r)) λ ()) (~ i) ] ⟩
+           [ 1 , (replicateFinVec 1 ( φ .fst 1r)) ++Fin (λ ()) ] ∎
+
+  N-ob ((snd (CompOpenDistLattice X) DistLatticeStr.∨l U) V) A x = U .N-ob A x ∨l V .N-ob A x
+    where instance _ = ZariskiLattice A .snd
+  N-hom ((snd (CompOpenDistLattice X) DistLatticeStr.∨l U) V)  {x = A} {y = B} φ = funExt path
+    where
+    instance
+      _ = ZariskiLattice A .snd
+      _ = ZariskiLattice B .snd
+    path : ∀ x → U .N-ob B (X .F-hom φ x) ∨l V .N-ob B (X .F-hom φ x)
+               ≡ 𝓛 .F-hom φ (U .N-ob A x ∨l V .N-ob A x)
+    path x = U .N-ob B (X .F-hom φ x) ∨l V .N-ob B (X .F-hom φ x)
+           ≡⟨ cong₂ _∨l_ (funExt⁻ (U .N-hom φ) x) (funExt⁻ (V .N-hom φ) x) ⟩
+             𝓛 .F-hom φ (U .N-ob A x) ∨l 𝓛 .F-hom φ (V .N-ob A x)
+           ≡⟨ sym (inducedZarLatHom φ .snd .pres∨l _ _) ⟩
+             𝓛 .F-hom φ (U .N-ob A x ∨l V .N-ob A x) ∎
+
+  N-ob ((snd (CompOpenDistLattice X) DistLatticeStr.∧l U) V) A x = U .N-ob A x ∧l V .N-ob A x
+    where instance _ = ZariskiLattice A .snd
+  N-hom ((snd (CompOpenDistLattice X) DistLatticeStr.∧l U) V)  {x = A} {y = B} φ = funExt path
+    where
+    instance
+      _ = ZariskiLattice A .snd
+      _ = ZariskiLattice B .snd
+    path : ∀ x → U .N-ob B (X .F-hom φ x) ∧l V .N-ob B (X .F-hom φ x)
+               ≡ 𝓛 .F-hom φ (U .N-ob A x ∧l V .N-ob A x)
+    path x = U .N-ob B (X .F-hom φ x) ∧l V .N-ob B (X .F-hom φ x)
+           ≡⟨ cong₂ _∧l_ (funExt⁻ (U .N-hom φ) x) (funExt⁻ (V .N-hom φ) x) ⟩
+             𝓛 .F-hom φ (U .N-ob A x) ∧l 𝓛 .F-hom φ (V .N-ob A x)
+           ≡⟨ sym (inducedZarLatHom φ .snd .pres∧l _ _) ⟩
+             𝓛 .F-hom φ (U .N-ob A x ∧l V .N-ob A x) ∎
+
+  DistLatticeStr.isDistLattice (snd (CompOpenDistLattice X)) = makeIsDistLattice∧lOver∨l
+    isSetNatTrans
+    (λ _ _ _ → makeNatTransPath (funExt₂
+                 (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∨lAssoc _ _ _)))
+    (λ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∨lRid _)))
+    (λ _ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∨lComm _ _)))
+    (λ _ _ _ → makeNatTransPath (funExt₂
+                 (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lAssoc _ _ _)))
+    (λ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lRid _)))
+    (λ _ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lComm _ _)))
+    (λ _ _ → makeNatTransPath (funExt₂ -- don't know why ∧lAbsorb∨l doesn't work
+               (λ A _ → ZariskiLattice A .snd .DistLatticeStr.absorb _ _ .snd)))
+    (λ _ _ _ → makeNatTransPath (funExt₂ -- same here
+                 (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧l-dist-∨l _ _ _ .fst)))
+
+  -- TODO: (contravariant) action on morphisms
```

Comment:
Please fix this as it should be easy, then we can merge

Reply:
Done